xarray/core/formatting.py

```diff
@@ -145,15 +145,50 @@ def format_array_flat(items_ndarray, max_width):
     return pprint_str
 
 
+def _get_level_values(index, level_num, max_size=50):
+    """Similar to `pd.MultiIndex.get_level_values` but
+    here truncated to `max_size`.
+    """
+    unique = index.levels[level_num]
+    labels = index.labels[level_num]
+    size = min(max_size, labels.size)
+    filled = pd.core.algorithms.take_1d(unique.values, labels[:size],
+                                        fill_value=unique._na_value)
+    _simple_new = unique._simple_new
+    values = _simple_new(filled, name=index.names[level_num],
+                         freq=getattr(unique, 'freq', None),
+                         tz=getattr(unique, 'tz', None))
+    return values
+
+
+def _summarize_index_level(name, index, col_width, max_width):
+    first_col = pretty_print('    - %s ' % name, col_width)
+    front_str = first_col + ('%s ' % index.dtype)
+    values_str = format_array_flat(index, max_width - len(front_str))
+    return front_str + values_str
+
+
 def _summarize_var_or_coord(name, var, col_width, show_values=True,
                             marker=' ', max_width=None):
     if max_width is None:
         max_width = OPTIONS['display_width']
     first_col = pretty_print('  %s %s ' % (marker, name), col_width)
     dims_str = '(%s) ' % ', '.join(map(str, var.dims)) if var.dims else ''
-    front_str = first_col + dims_str + ('%s ' % var.dtype)
+    index = var.to_index()
```

Comment:
Yeah, that seems OK.

Probably better to first do the indexing up to `max_size`, though. Otherwise this could load a lot more data into memory than necessary when used with dask.